File: projects/01-promo-map/backend/schemas/favorite.py
"""즐겨찾기 스키마"""
from pydantic import BaseModel

__all__ = ["FavoriteRequest", "FavoriteResponse"]


class FavoriteRequest(BaseModel):
    store_id: int


class FavoriteResponse(BaseModel):
    id: int
    store_id: int
    store_name: str
    store_brand: str
    store_category: str
    icon_color: str
    icon_letter: str

    class Config:
        from_attributes = True

File: projects/01-promo-map/flutter_app/lib/data/models/favorite.dart
class Favorite {
  final int id;
  final int storeId;
  final String storeName;
  final String storeBrand;
  final String storeCategory;
  final String iconColor;
  final String iconLetter;

  const Favorite({
    required this.id,
    required this.storeId,
    required this.storeName,
    required this.storeBrand,
    this.storeCategory = 'general',
    this.iconColor = '#FF6B35',
    this.iconLetter = 'S',
  });

  factory Favorite.fromJson(Map<String, dynamic> json) {
    return Favorite(
      id: json['id'] as int,
      storeId: json['store_id'] as int,
      storeName: json['store_name'] as String,
      storeBrand: json['store_brand'] as String,
      storeCategory: (json['store_category'] as String?) ?? 'general',
      iconColor: (json['icon_color'] as String?) ?? '#FF6B35',
      iconLetter: (json['icon_letter'] as String?) ?? 'S',
    );
  }
}
